roles/etcd/tasks/certificates/remove_generated_certificates.yml

```diff
@@ -1,5 +1,11 @@
 ---
+- name: Determine if generated etcd certificates exist
+  stat:
+    path: "{{ etcd_conf_dir }}/generated_certs"
+  register: etcd_generated_certs_dir_stat
+
 - name: Remove generated etcd certificates
   file:
     path: "{{ etcd_conf_dir }}/generated_certs"
     state: absent
+  when: etcd_generated_certs_dir_stat.stat.exists | bool
```

Comment:
It the condition necessary? The `absent` should by idempotent. If the directory exists, it get deleted, if it does not exist, nothing will happen. For that reason, I removed this piece of code intentionally.

Reply:
You're right, I'll take this out.